examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+   "execution_count": null,
+   "id": "de17b188",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%pip install ipyreact\n",
+    "# This line is for JupyterLite (in case that this takes long, restart the kernel and run this cell again)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d85f911",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext ipyreact"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c70f7253-98d5-4131-bf6b-1ac3433e2985",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%react\n",
+    "\n",
+    "import * as React from \"react\";\n",
+    "\n",
+    "export default function MyButton() {\n",
+    "    return ( < button > X < /button>);\n",
+    "    }"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "4b51d34f-8abc-4b7b-936e-6d582a1b64fe",
+   "metadata": {},
+   "source": [
+    "This works like a charm!  \n",
+    "Next, we **convert this into a widget.**  \n",
+    "For that, we need the code in a _ems string inside a class that inherits from `ipyreact.ReactWidget`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7c70385c-638e-482b-80e1-6f27614da867",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "\n",
+    "class MyExampleWidget(ipyreact.ReactWidget):\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function MyButton() {\n",
+    "        return < button > X < /button> \n",
+    "    };\"\"\"\n",
+    "\n",
+    "MyExampleWidget()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "35f22aef-50c2-4772-b9fa-44f5fb486ee6",
+   "metadata": {},
+   "source": [
+    "now it's time to make some **styling using CSS**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38e8afeb-274c-4c62-99a7-6200fc9c179d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "css_rules = \"\"\"\n",
+    "button {\n",
+    "    background-color: yellow;\n",
+    "    border-radius: 10px;\n",
+    "}\n",
+    "\"\"\"\n",
+    "HTML(\"<style>\" + css_rules + \"</style>\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cb19480f-b3c6-44be-968c-06edf944d42f",
+   "metadata": {},
+   "source": [
+    "You can also load CSS from a file.  \n",
+    "And you can add extra CSS without overwriting the previous CSS import."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c4a644e0-1dbe-4f50-9bb5-5a54f1fedbaf",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "from pathlib import Path\n",
+    "css_rules = Path(\"styles_orange.css\").read_text()\n",
+    "HTML(\"<style>\" + css_rules + \"</style>\")"
```

Comment:
Maybe an explanation along the lines of

Inline JS and CSS is great for examples and playing around.  For more serious development you probably want to use a text editor on separate files that are automatically reloaded.  This way you get the normal syntax hilighting that your familiar with.

Reply:
I see that this is mentioned at the end.  I will draft an introduction paragraph that mentions this piece.

Another thing to mention here, or somewhere else in the tutorial is:
"Normal development practice is to have each resource (css, js, python) in a separate file.  There are advantages to writing inline in jupyter notebook cells.  It's much quicker to get started, and examples are more compact with less file switching to follow along.  Because of the self containted literate programming aspect, this tutorial will use the inline method.

Reply:
I incorporated that in the end of the example.